### PR TITLE
[Fix-9921] Add the mechanism to increment the workerWaitingTaskCount in HeartBea…

### LINKED ISSUE #9921: [Bug] [Server] Worker workerWaitingTaskCount has always been zero
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/dolphinscheduler/issues?q=is%3Aissue) and found no similar issues.

### What happened

[zk: localhost:2181(CONNECTED) 25] get /dolphinscheduler/nodes/worker/default/host:1234
0.02,0.39,0.38,19.06,16.0,0.3,1651832787990,1651833001883,0,30984,100,2,0

image

### What you expected to happen

zk data of the last parameter should have a value

The reason is that workerWaitingTaskCount cannot get variables outside the thread
Solution: It is recommended to disassemble HeartBeatTask into MasterHeartBeatTask and Worker HeartBeatTask

### How to reproduce

dev

### Anything else

_No response_

### Version

3.0.0-alpha

### Are you willing to submit PR?

- [x] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by netemadi**
## Purpose of the pull request
Add the mechanism to increment the workerWaitingTaskCount inside HeartBeat class to fix #9921

## Brief change log
When HeartBeatTask is run priodically from heartBeatExecutor, then it will increment the workerWaitingTaskCount

## Verify this pull request

This pull request is code cleanup without any test coverage.

### CODE REVIEW

File: dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/registry/HeartBeatTask.java [hunk 1]
```diff
@@ -67,6 +67,15 @@ public HeartBeatTask(long startupTime,
         this.heartBeat = new HeartBeat(startupTime, maxCpuloadAvg, reservedMemory, hostWeight, workerThreadCount);
     }
 
+    public void setWorkerWaitingTaskCount(int count){
+        this.workerWaitingTaskCount  = count;
+    }
+
+    public void incrementWorkerWaitingTaskCount(){
+        this.workerWaitingTaskCount++;
```
Comment (brave-lee) [inline dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/registry/HeartBeatTask.java:75]: I don't think this modification can solve this problem, this value is the number of tasks to be run held by the current worker, not the count of heartbeat

Comment (brave-lee) [reply dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/registry/HeartBeatTask.java:75]: I suggest splitting this class into master and worker to obtain it separately. BTW, please test it fully

**Comment by zhongjiajie** [comment]
I restarted the failed CI

File: dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/registry/HeartBeatTask.java (hunk 1 above)
Comment (ruanwenjun) [reply dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/registry/HeartBeatTask.java:75]: +1， this patch cannot solve this problem, you can use a Supplier to replace the workerWaitingTaskCount to solve this.

**Comment by caishunfeng** [comment]
close due to #11232

**Pull request closed without merging**